gtk: pass through keypress when clipboard has no text

cyppe opened the pull request:
## Summary

When `paste_from_clipboard` is triggered but the clipboard contains no text (e.g., an image), the action now returns `false` to indicate it couldn't be performed. This enables the `performable:` keybind prefix to work correctly for paste actions.

## Problem

On GTK/Linux, when a user has `keybind = ctrl+v=paste_from_clipboard` and the clipboard contains an image (not text), pressing Ctrl+V does nothing. Applications like `opencode` that handle their own clipboard reading via `wl-paste` never receive the keypress.

## Solution

Make `clipboardRequest` return `bool` to indicate whether the action could be performed. For paste requests on GTK, synchronously check if the clipboard contains text formats before starting the async read. When no text format is available, return `false`.

Users can now use:
```
keybind = performable:ctrl+v=paste_from_clipboard
```

When the clipboard has no text, the keybind is not consumed and Ctrl+V passes through to the terminal application.

## Changes

- `Surface.startClipboardRequest` now returns `bool`
- `paste_from_clipboard` / `paste_from_selection` actions return the result
- GTK apprt checks clipboard formats synchronously before async read
- Embedded apprt always returns `true` (can't check synchronously)

## Testing

1. Add `keybind = performable:ctrl+v=paste_from_clipboard` to config
2. Copy an image to clipboard
3. Open an application that handles image paste (e.g., `opencode`)
4. Press Ctrl+V
5. Image pastes successfully (app receives keypress and handles clipboard itself)

## Disclaimer

Most of the changes is done with Opus 4.5

cyppe committed:
gtk: pass through keypress when clipboard has no text

When paste_from_clipboard is triggered but the clipboard contains no
text (e.g., an image), send the raw Ctrl+V keypress to the terminal
instead of silently returning. This allows applications to handle
their own clipboard reading (e.g., via wl-paste for images on Wayland).

jcollie reviewed (changes requested):
This should be taking advantage of the `performable` prefix and not relying on sending a fixed `ctrl+v` if there is no text in the buffer.

cyppe committed:
gtk: support performable keybinds for clipboard paste

Make clipboardRequest return bool to indicate whether the action could
be performed. For paste requests, synchronously check if the clipboard
contains text formats before starting the async read.

This allows 'performable:paste_from_clipboard' keybinds to pass through
when the clipboard contains non-text content (e.g., images), enabling
terminal applications to handle their own clipboard reading.

Changes:
- Surface.startClipboardRequest now returns bool
- paste_from_clipboard/paste_from_selection actions return the result
- GTK apprt checks clipboard formats synchronously before async read
- Embedded apprt always returns true (can't check synchronously)
- All other call sites discard the return value with _

mitchellh reviewed (changes requested):
@jcollie is right. This is the wrong approach.

The correct approach is to return false from our apprt action so our performable infrastructure works. The issue is GDK clipboard reads are async and we require a response immediately. 

The question is if GDK exposes an API to at least ask "are you empty" synchronously. And it may, but not sure if it works as expected, e.g. https://docs.gtk.org/gdk4/method.Clipboard.get_content.html

macOS clipboards are also async, but we're able to ask the empty question synchronously.

mitchellh reviewed (commented):
This looks better. I haven't verified if it works yet, and I don't love hunting for specific mime types like that but maybe there is no matter way.

cyppe commented:
I am no expert, but the final commit is verified to work. Updated PR description too.

Have not run any test suite, so cannot say if something else broke down.

cyppe commented:
<img width="2374" height="728" alt="image" src="https://github.com/user-attachments/assets/8bbf6a3a-e8dc-44a0-8306-c72344a17cb5" />

Some attempts incoming, will test it.

cyppe committed:
Improve type detection

mitchellh reviewed (changes requested):
Looks great. One more request.

cyppe committed:
review

mitchellh reviewed (approved):
Approved pending CI. Thank you.

cyppe commented:
@mitchellh question - I have friends telling my that on mac they  need to use ctrl+v to paste images to tools like opencode using default keybinds even if all other paste is opt+v.

Tried to unbind opt+v and bind paste_from_clipboard to some random key to see if it can make ghostty trigger native "ctrl+v" style paste on opt+v. But it doesnt.

keybind = opt+v=unbind
keybind = opt+n=paste_from_clipboard

Any idea if there is some feature already in ghostty to get native "ctrl+v" paste to trigger on osx on opt+v?

Otherwise - some pointers to what concept to use if I would attempt a PR, as performable doesnt cover it as it doesnt fall through to native paste on opt+v. 

So it basically needs:

```
keybind = performable:opt+v=paste_from_clipboard + something to trigger native paste when not performable
```
